src/set.rs

```diff
@@ -422,6 +422,56 @@ where
         (index, existing.is_none())
     }
 
+    /// Insert the value into the set at its ordered position among values
+    /// sorted by `cmp`.
+    ///
+    /// This is equivalent to finding the position with
+    /// [`binary_search_by`][Self::binary_search_by], then calling
+    /// [`insert_before`][Self::insert_before].
+    ///
+    /// If the sorted item is found in the set, it returns the index of that
+    /// existing item and `false`. Otherwise, it inserts the new item and
+    /// returns its sorted index and `true`.
```

Comment:
This paragraph needs adjustment because existing items also move -- or maybe just leave it out like you did with the map method.